net/instaweb/rewriter/webscale_make_scripts_async.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ *  COPYRIGHT:
+ *     (C) Copyright Webscale Networks 2017.
+ *      Licensed Materials - All Rights Reserved.
+ *    
+*/
+
+#include <vector>
+
+#include "base/logging.h"
+#include "net/instaweb/rewriter/public/rewrite_driver.h"
+#include "net/instaweb/rewriter/public/rewrite_driver_factory.h"
+#include "net/instaweb/rewriter/public/webscale_make_scripts_async.h"
+#include "pagespeed/kernel/html/html_element.h"
+#include "pagespeed/kernel/html/html_name.h"
+#include "pagespeed/kernel/html/html_node.h"
+#include "pagespeed/kernel/base/message_handler.h"
+#include "pagespeed/kernel/base/statistics.h"
+#include "pagespeed/kernel/base/string.h"
+#include "pagespeed/kernel/base/string_util.h"
+#include "pagespeed/kernel/util/re2.h"
+
+namespace net_instaweb {
+
+WebscaleMakeScriptsAsync::WebscaleMakeScriptsAsync(RewriteDriver* rewrite_driver, MessageHandler* message_handler)
+    : CommonFilter(rewrite_driver) {
+  Statistics* statistics = rewrite_driver->statistics();
+  message_handler = driver()->message_handler();
+  escaped_urls = ConstructPatternFromCustomUrls(rewrite_driver->options());
+}
+
+WebscaleMakeScriptsAsync::~WebscaleMakeScriptsAsync(){
+}
+
+void WebscaleMakeScriptsAsync::InitStats(Statistics* statistics) {
+}
+
+
+void WebscaleMakeScriptsAsync::StartDocumentImpl() {
+}
+
+
+void WebscaleMakeScriptsAsync::StartElementImpl(HtmlElement* element) {
+  message_handler = driver()->message_handler();
+
+  // Script element found.
+  if (element->keyword() == HtmlName::kScript) {
+    const char* src_attribute = element->EscapedAttributeValue(HtmlName::kSrc);
+    // Script element has a 'src' attribute.
+    if (src_attribute != NULL) {
+      // Convert the src element to a GoogleString format for compatibility.
+      GoogleString* src_string = new GoogleString(src_attribute);
```

Comment:
This is a memory leak: you allocate the string but never free it.  This will slowly accumulate cruft in pagespeed until the ADC runs out of memory.

Since you don't use the string outside of this function, don't use `new`, just use

```c++
GoogleString src_string(src_attribute);
```

with `src_string.c_str()` below.... except that you don't use anything but `c_str` below, which just returns a `const char*`... so you can just use `src_attribute` directly and throw `src_string` on the ground.